products/cloudscheduler/terraform.yaml

```diff
@@ -0,0 +1,46 @@
+# Copyright 2018 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Provider::Terraform::Config
+name: CloudScheduler
+overrides: !ruby/object:Provider::ResourceOverrides
+  Job: !ruby/object:Provider::Terraform::ResourceOverride
+    properties:
+      name: !ruby/object:Provider::Terraform::PropertyOverride
+        custom_expand: 'templates/terraform/custom_expand/cloud_scheduler_job_name.erb'
+        custom_flatten: 'templates/terraform/custom_flatten/name_from_self_link.erb'
+      httpTarget.body: !ruby/object:Provider::Terraform::PropertyOverride
+        custom_expand: 'templates/terraform/custom_expand/base64_encode_body.erb'
```

Comment:
I'm not married to these custom expanders/flatteners and didn't realise there was built in functions, I'll use those.

Reply:
just been reading, I see now that the user would be required to wrap the body data in the function, .i.e., 

```
resource "google_cloud_scheduler_job" "job" {
	name     = "%s"
	description = "test job"
	schedule = "*/2 * * * *"

	pubsub_target = {
		topic_name = "projects/%s/topics/build-triggers"
		data = "${base64encode("test")}"
	}
}
```

is this the correct assumption? I think my approach was to take the complexity out of the user base64 encoding data themselves and having the data decoded when viewing the plan to avoid confusion.

I could add unit testing for these methods, I think there is value in keeping this functionality but understand if this goes against any precedent and can remove

Reply:
Any further thoughts on this @emilymye ?

Reply:
I'm back and forth on this. I think I'd be more comfortable with having the state value be base64-encoded rather than having a large text-blob with weird characters floating around in TF state.

Reply:
I've changed that now, removed the custom base64 flatten/expand and changed the test to use built in function